### ISSUE TITLE
running s_server from openssl compiled with no-ecdh results in EC related error

### ISSUE THREAD

**Issue by vladak** [openssl/openssl#3303]
Compiled code from branch `OpenSSL_1_0_2-stable` with `no-ecdh` configure option and running `s_server`:

```
openssl s_server -ssl3 -cert mycert.pem -key mykey.pem -debug -www -port 4433
Error setting EC curve
```

This is caused by the `no_ecdhe` variable in `s_sever.c` to be 0 implicitly. Once passed to `args_ssl_call()` it will cause this error to be thrown:

```
1585      /*
1586       * This is a special case to keep existing s_server functionality: if we
1587       * don't have any curve specified *and* we haven't disabled ECDHE then
1588       * use P-256.
1589       */
1590      if (!no_ecdhe) {
1591          if (SSL_CONF_cmd(cctx, "-named_curve", "P-256") <= 0) {
1592              BIO_puts(err, "Error setting EC curve\n");
1593              ERR_print_errors(err);
1594              return 0;
1595          }
1596      }
```

The `no_ecdhe` variable in `s_sever.c` should probably be set to 1 if compiling with the `OPENSSL_NO_ECDH` define.

Ironically, the workaround is to use the `-no_ecdhe` option for `s_server` which is however commented out:

```
$ openssl s_server -h 2>&1 | grep ecdh
<nothing>
```

**Comment by gopinatharjuan** [comment]
We are also facing the same issue kindly advice.

**Comment by mattcaswell** [comment]
This was issue was fixed in recent versions of OpenSSL by commit 8caab744f56. It won't be fixed in the 1.0.2 branch since that is in security-fix only mode.
